src/missing_deps.cc

```diff
@@ -0,0 +1,160 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "missing_deps.h"
+
+#include <stdio.h>
+#include <string.h>
+
+#include "depfile_parser.h"
+#include "deps_log.h"
+#include "disk_interface.h"
+#include "graph.h"
+#include "state.h"
+#include "util.h"
+
+MissingDependencyScannerDelegate::~MissingDependencyScannerDelegate() {}
+
+void MissingDependencyPrinter::OnMissingDep(Node* node, const std::string& path,
+                                            const Rule& generator) {
+  printf("Missing dep: %s uses %s (generated by %s)\n", node->path().c_str(),
```

Comment:
You might consider using iostreams, instead of printf. Printf is so.... ugh. For lack of better terminology to use.

Reply:
Fully agree on the terminology -- I used printf solely based on the fact that there are 0 (zero) usages of iostreams across ninja right now. If iostreams are "okay" (especially in a tool) then sure,

Reply:
While there's zero use of iostreams in terms of `std::cout`, there are several places where std::stringstream is used to construct a string (which, in a few situations, is then passed to fprintf...?)

While I am not an official project maintainer, and therefore one of the official maintainers can just overrule me, personally I think (for all it's warts...) iostreams is miles and miles ahead of fprintf, not the least reason of which is that it doesn't explicitly require nul-terminated strings.

Reply:
Let's see how it goes with std::cout.